### DISCUSSION TITLE
signOut() not working

### DISCUSSION THREAD

**Discussion by komposer-aml** [supabase/supabase#9639] [Questions] [answered]
Hello,

I am calling supabase.auth.signOut() upon a button click. However, it doesn't seem to be doing anything.

Here are the console logs for signOut() and useUser() (next.js/React auth helper)

In my Auth logs, I am seeing logout events returning status code 204, but useUser is still returning with an active session after calling signOut().

How do I signOut users on Supabase?
<img width="430" alt="Screen Shot 2022-10-18 at 3 05 14 AM" src="https://user-images.githubusercontent.com/41843630/196361053-8b2dfa4c-4c9a-43e5-8ade-15e0037f01ce.png">

**Comment by GaryAustin1** [comment, marked as answer]
The current JWT in the client is valid until it expires even if signed out. It just can’t be refreshed with the refresh token,

From the signOut docs:   
For server-side management, you can revoke all refresh tokens for a user by passing a user's JWT through to auth.api.signOut(JWT: string). There is no way to revoke a user's access token jwt until it expires. It is recommended to set a shorter expiry on the jwt for this reason.

**Comment by tteguayco** [comment]
Strange thing is that, when executing `supabase.auth.signOut()`, it hangs there and the execution never comes back to the caller.

I have no issues in a previous project built with Supabase but for this new one I'm working on I keep facing this issue.

**Reply by GaryAustin1** [reply]
Might be this. https://github.com/supabase/gotrue-js/issues/762

**Reply by tteguayco** [reply]
Thanks a lot @GaryAustin1 for pointing to that thread.

Inspired by one of the answers at the end, I replace this piece:
```
const { data: authListener } = supabaseClient.auth.onAuthStateChange(async () => {
    const res = await supabaseClient.auth.getSession();
    setUser(res.data.session?.user);
    setAccessToken(res.data.session?.access_token);
});
```
by this other:
```
const { data: authListener } = supabaseClient.auth.onAuthStateChange(async (_, session) => {
    setUser(session?.user);
    setAccessToken(session?.access_token);
});
```

And now it's working, even when I don't fully understand the change. Also I don't understand the fact that my previous project is working with the first snippet and the versions of the ```@supabase/supabase-js``` dependency is ```2.2.0``` for both projects.
